rosette/src/Export.cc

```diff
@@ -0,0 +1,185 @@
+#include "Ob.h"
+#include "Code.h"
+#include "Prim.h"
+#include "Number.h"
+
+#include "CommandLine.h"
+
+#include <google/protobuf/text_format.h>
+#include <Ob.pb.h>
+#include <fstream>
+
+
+#include <string>
+
+ObjectCodePB::ObjectCode objectCode;
+
+void collectExportCode(Code *code) {
+    if ('\0' == *ExportFile)
+        return;
+
+    if (VerboseFlag) fprintf(stderr, "\n%s\n", __PRETTY_FUNCTION__);
+
+    // Build the code block protobuf to export
+    ObjectCodePB::CodeBlocks * cbs = objectCode.mutable_codeblocks();
+    ObjectCodePB::CodeBlock * cb = cbs->add_codeblock();
+
+    // Get access to the codevec and litvec to be exported.
+    CodeVec * codevec = code->codevec;
+    Tuple * litvec = code->litvec;
+
+    // Save the binary opCodes from the codevec
+    const char * code_binary = (char *)codevec->absolutize(0);
+    size_t code_size = codevec->numberOfWords() * sizeof(uint32_t);
+    ObjectCodePB::CodeVec * cv = cb->mutable_codevec();
+    cv->set_opcodes( std::string(code_binary, code_size) );
+
+    // Create the export litvec
+    ObjectCodePB::LitVec * lv = cb->mutable_litvec();
+
+    // Add objects or references to them to the export litvec
+    for (int i = 0; i < litvec->numberOfElements(); i++) {
+        pOb ob = litvec->elem(i);
+        std::string type = BASE(litvec->elem(i))->typestring();
+        ObjectCodePB::LitVecItem *item = lv->add_item();
+
+        if (type == "Actor") {
+            ObjectCodePB::Object * newOb = item->mutable_ob();
```

Comment:
Factor out creating newOb.

Reply:
P.S. pull from dev; your branch is out of date.

Reply:
I did it this way on purpose so that some objects can be populated into the litvec item directly, and others can be accessed via the central object list.